magefiles/installation/install.go

```diff
@@ -149,6 +164,149 @@ func (i *InstallAppStudio) cloneInfraDeployments() (*git.Remote, error) {
 	return repo.CreateRemote(&config.RemoteConfig{Name: i.LocalForkName, URLs: []string{fmt.Sprintf("https://github.com/%s/infra-deployments.git", i.LocalGithubForkOrganization)}})
 }
 
+func MergePRInRemote(branch string, forkOrganization string, repoPath string) error {
+	if branch == "" {
+		klog.Fatal("The branch for upgrade is empty!")
+	}
+
+	repo, err := git.PlainOpen(repoPath)
+	if err != nil {
+		klog.Fatal(err)
+	}
+
+	branches, err := repo.Branches()
+	if err != nil {
+		klog.Fatal(err)
+	}
+
+	var branchRepo string
+	err = branches.ForEach(func(ref *plumbing.Reference) error {
+		if !strings.Contains("main", ref.Name().String()) {
+			branchRepo = ref.Name().String()
+		}
+		return nil
+	})
+	if err != nil {
+		klog.Fatal(err)
+	}
+
+	wt, err := repo.Worktree()
+	if err != nil {
+		klog.Fatal(err)
+	}
+
+	err = wt.Checkout(&git.CheckoutOptions{
+		Branch: plumbing.ReferenceName(branchRepo),
+	})
+	if err != nil {
+		klog.Fatal(err)
+	}
+
+	if forkOrganization == "" {
+		_, err = exec.Command("git", "-C", repoPath, "merge", "remotes/origin/"+branch, "-q").Output()
+	} else {
+		repoURL := fmt.Sprintf("https://github.com/%s/infra-deployments.git", forkOrganization)
+		_, err = repo.CreateRemote(&config.RemoteConfig{
+			Name: "forked_repo",
+			URLs: []string{repoURL},
+		})
+		if err != nil {
+			klog.Fatal(err)
+		}
+
+		err = repo.Fetch(&git.FetchOptions{
+			RemoteName: "forked_repo",
+		})
+		if err != nil {
+			klog.Fatal(err)
+		}
+
+		_, err = exec.Command("git", "-C", repoPath, "merge", "remotes/forked_repo/"+branch).Output()
+	}
+	if err != nil {
+		klog.Fatal(err)
+	}
+
+	var auth = &http.BasicAuth{
+		Username: "123",
+		Password: utils.GetEnv("GITHUB_TOKEN", ""),
+	}
+	err = repo.Push(&git.PushOptions{
```

Comment:
I was stuck for quite some time here trying to figure out why the push is failing with `non-fast-forward update`. Turns out it's trying to push `main` branch instead of the `preview-` one. Once I added `RefSpecs` pushing the right branch here, everything woked fine.

Reply:
```
err = repo.Push(&git.PushOptions{
  RefSpecs:   []config.RefSpec{config.RefSpec(fmt.Sprintf("%s:%s", branchRepo, branchRepo))},
  RemoteName: "qe",
  Auth:       auth,
})
```
This is what I have here now (not sure it's the best what it could be)

Reply:
Thanks! Added.